lib/meataxe.gi

```diff
@@ -518,17 +514,25 @@ SMTX.SubmoduleGModule:=SMTX.SubGModule;
 ##  See the description for 'SMTX.InducedAction' for
 ##  description of the matrices
 ##
-SMTX.SubQuotActions:=function(matrices,sub,dim,subdim,F,typ)
-local s, c, q, leadpos, zero, zerov, smatrices, newg, im, newim, k, subi,
+SMTX.SubQuotActions:=function(module,sub,typ)
+local matrices, dim, subdim, F,
+      s, c, q, leadpos, zero, zerov, smatrices, newg, im, newim, k, subi,
       qmats, smats, nmats, sr, qr, g, h, erg, i, j;
 
+  dim:=SMTX.Dimension(module);
+  F:=SMTX.Field(module);
+  matrices:=module.generators;
+
+  Assert(0, dim = Length(sub[1]));
```

Comment:
We already use `Assert(0,` in about 30 spots in the library. It is exactly as you say. In this cases I added this as a kind of "paranoia check" and I wanted it to be always one because a frustratingly big patch of the library is not exercised by our test suite.